src/zarr/v3/metadata.py

```diff
@@ -115,7 +115,7 @@ class ArrayMetadata(Metadata):
     chunk_grid: ChunkGrid
     chunk_key_encoding: ChunkKeyEncoding
     fill_value: Any
-    codecs: CodecPipeline
+    codecs: BatchedCodecPipeline
```

Comment:
Does this mean the use the batched pipeline by default always? That might not be the worst thing, but wondering if it might be useful to implement some mechanism to pass a policy to `parse_codecs` that could allow implementing an `InterleavedCodecPipeline` or something down the line?

Reply:
I definitely want to add a `InterleavedCodecPipeline` as well. We could make that configurable through the runtime configuration.